src/proxy.c

```diff
@@ -277,12 +273,26 @@ cc_proxy_msg_read (GIOChannel *source, GIOCondition condition,
 		}
 	}
 
-	/* ensure DO NOT copy extra bytes */
-	bytes_to_copy = MIN((size_t)bytes_read-data_offset, (size_t)p_msg.length);
-	g_string_append_len(proxy_data->msg_received,
-		iov_buffer+data_offset, (ssize_t)bytes_to_copy);
-
-	if (bytes_read >= p_msg.length+data_offset) {
+	/* cc-proxy sends a fd as out-of-band data
+	 * followed by an 'F'.
+	 * *DO NOT* copy these characters to msg_received
+	 */
+	if (bytes_read > 1 || iov_buffer[0] != 'F') {
```

Comment:
so, when bytes_read is > 1  iov_buffer[0] will never be 'F'  (well will be too big ), right? 
lgtm